### DISCUSSION TITLE
importing extensions for the base Prisma client

### DISCUSSION THREAD

**Discussion by ReeceBennett** [prisma/prisma#20083] [Q&A]
Im trying to find out if i have to create the extensions and add them in each service.ts file or if i can somehow call them in the schema and have the extenstions apply themself to the default "Prisma/client"

if i cant i will need to go to all service modules / controllers and import the customized Prisma export

**Comment by nurul3101** [comment]
Hello @ReeceBennett 👋 

> create the extensions and add them in each service.ts

You shouldn't need to define client extensions in each file, ideally you need to define it only once when you are initialising your Prisma Client. Here are few [examples](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions/extension-examples) that might be helpful.

> somehow call them in the schema

Can you please elaborate on this? Client Extensions cannot be invoked from schema file and generally there shouldn't be any need as well. Do you mind elaborating on your use case?

**Reply by nurul3101** [reply]
Also, do you mind elaborating on how you are instantiating your `PrismaClient`?
Are you instantiating `PrismaClient` in each `service.ts` file?

**Reply by ReeceBennett** [reply]
Hi Nurul, I appreciate you responding 

`
import { PrismaClient } from "@prisma/client"; 
const prisma = new PrismaClient();
`

the Above is in all of my service.ts i have a service for each section in my db , Customers, addresses, people etc.... 
I was hoping that the extenstions would be included in the default prisma, since posting i have just slowly made my way through my application changing to a globalextended.prisma.ts

thank you for taking your time to respond I appreciate it alot

**Reply by nurul3101** [reply]
Thanks for getting back and letting us know. My recommendation would be to use a single PrismaClient instance, if possible, which you seem to be doing now. 
Here's our [recommendation](https://www.prisma.io/docs/guides/performance-and-optimization/connection-management#prismaclient-in-long-running-applications) in context of using a single PrismaClient Instance.

**Reply by nurul3101** [reply]
It would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 
Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by nurul3101** [reply]
We are closing this Discussion due to inactivity. Please open a new Discussion if you have a similar problem 🙏
